crates/trie/src/trie_cursor/subnode.rs

```diff
@@ -75,48 +71,34 @@ impl CursorSubNode {
 
     /// Returns `true` if the state flag is set for the current nibble.
     pub fn state_flag(&self) -> bool {
-        if let Some(node) = &self.node {
-            if self.nibble >= 0 {
-                return node.state_mask.is_bit_set(self.nibble as u8)
-            }
-        }
-        true
+        self.node
+            .as_ref()
+            .map_or(true, |node| self.nibble >= 0 && node.state_mask.is_bit_set(self.nibble as u8))
```

Comment:
previously, if `self.nibble < 0` then method would return `true`
```suggestion
            .map_or(true, |node| self.nibble < 0 || node.state_mask.is_bit_set(self.nibble as u8))
```